salt/returners/local_cache.py

```diff
@@ -100,12 +100,13 @@ def prep_jid(nocache=False, passed_jid=None, recurse_count=0):
 
     # Make sure we create the jid dir, otherwise someone else is using it,
     # meaning we need a new jid.
-    try:
-        os.makedirs(jid_dir_)
-    except OSError:
-        time.sleep(0.1)
-        if passed_jid is None:
-            return prep_jid(nocache=nocache, recurse_count=recurse_count+1)
+    if not os.path.isdir(jid_dir_):
```

Comment:
This would be more correct as the following:

    # Make sure we create the jid dir, otherwise someone else is using it,
    # meaning we need a new jid.
    try:
        os.makedirs(jid_dir_)
    except OSError as exc:
        if exc.errno != errno.EEXIST:
            err = 'prep_jid failed creating jid directory: {0}'.format(exc)
            log.error(err)
            raise salt.exceptions.SaltCacheError(err)
        if passed_jid is None:
            # The JID that was just generated is already in use so get a new one.
            time.sleep(0.1)
            return prep_jid(nocache=nocache, recurse_count=recurse_count+1)